### PR TITLE
Fix tail truncation behavior when truncation rect is longer than last line rect

### PR DESCRIPTION
**Pull request by ocrickard**
This function assumed that the truncation token would be smaller than the last line.  This corrects the behavior for forced-truncation (maximum number of lines) cases where the truncation token is larger than the last line of text.

Fixes https://github.com/facebook/componentkit/issues/238

### CODE REVIEW

**Comment by ocrickard** [comment]
Somehow it's not finding my snapshot.  Will update.

**Comment by ocrickard** [comment]
Uggh, I had to download the iOS 8.1 simulator and re-record the snapshots for both iOS 8.1 and 8.3 in 64-bit and non-64-bit.

File: ComponentTextKit/TextKit/CKTextKitTailTruncater.mm [hunk 1]
```diff
@@ -95,10 +95,12 @@ - (NSUInteger)_calculateCharacterIndexBeforeTruncationMessage:(NSLayoutManager *
   NSUInteger firstClippedGlyphIndex = [layoutManager glyphIndexForPoint:beginningOfTruncationMessage
                                                         inTextContainer:textContainer
                                          fractionOfDistanceThroughGlyph:NULL];
+  // If it didn't intersect with any text then it should just return the last visible character index, since the
+  // truncation rect can fully fit on the line without clipping any other text.
+  if (firstClippedGlyphIndex == NSNotFound) {
+    return [layoutManager characterIndexForGlyphAtIndex:lastVisibleGlyphIndex];
+  }
   NSUInteger firstCharacterIndexToReplace = [layoutManager characterIndexForGlyphAtIndex:firstClippedGlyphIndex];
-  CKAssert(firstCharacterIndexToReplace != NSNotFound,
```
Comment (ocrickard) [inline ComponentTextKit/TextKit/CKTextKitTailTruncater.mm]: It's actually fine for this to be NSNotFound, it just means we don't intersect with the last line.

File: ComponentTextKitApplicationTests/CKTextComponentTests.mm [hunk 2]
```diff
@@ -384,4 +384,27 @@ - (void)testComplexLigatures
   CKSnapshotVerifyComponent(c, kUnrestrictedSize, @"");
 }
 
+- (void)testTruncationStringLargerThanLastLine
```
Comment (modocache) [inline ComponentTextKitApplicationTests/CKTextComponentTests.mm]: Can you rename this test to indicate what *should* happen when the truncation string is larger than the last line?

**Comment by modocache** [comment]
Accepted; just the one nit for the test method name.

**Pull request merged**